Mathlib/Data/List/Basic.lean

```diff
@@ -395,6 +395,11 @@ theorem cons_eq_append_iff {a b c : List α} {x : α} :
 @[deprecated] alias append_right_cancel := append_cancel_right -- deprecated since 2024-01-18
 #align list.append_right_cancel List.append_cancel_right
 
+@[simp] theorem append_left_cancel_nil_left {x y : List α}: x ++ y = x ↔ y = [] := by
+  constructor <;> intro h; apply List.append_cancel_left (as := x); all_goals simp[h]
+@[simp] theorem append_left_cancel_nil_right {x y : List α}: x = x ++ y ↔ y = [] :=
```

Comment:
Please put a blank line between two theorems. This makes files *so* much easier to read.